Add component that displays ETH address with etherscan link

banciur opened the pull request.

banciur committed:
Add component that displays ETH address with etherscan link

recmo commented on src/components/EthAddress.js:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  let ethScanPrefix;
+  switch (networkId) {
```
This is too much logic, it should be factored out of the view. The view should take an `addressLookupUrl` function that either returns an url or `null`. This function should be provided using a `getAddressLookupUrl` selector in `src/reducers/web3.js`, possibly using `src/chains.json` as a source.

recmo commented on src/components/EthAddress.js:
```diff
@@ -0,0 +1,49 @@
+};
+
+ETHAddress.propTypes = {
+  ethAddress: PropTypes.string.isRequired,
```
We can make a [custom validator](https://www.ian-thomas.net/custom-proptype-validation-with-react/) for Ethereum addresses.

recmo reviewed (commented):
This is good for a first version. In the next version we should have:

* Factor logic out of view
* Verify address correctness.
* Add checksums to addresses.
* Render addresses in a readable monospaced font
* Elide long addresses (best done using [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow))
* Full address in tooltip.
* Bonus points if you elide in the middle, like `0x7B9Bc4…2Ae47D`

The question is, do we want to do all this as part of this PR or as a next version?

banciur committed:
Fixes to EthAddress component based on pull request comments

Only "Factor logic out of view" was partly implemented.
Rest was left as TODO comments

recmo committed:
Merge branch 'master' into component_eth_address
